NetworkingCodingExcersise/NetworkingCodingExcersise/Program.cs

```diff
@@ -0,0 +1,122 @@
+﻿using System;
+using System.IO;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace NetworkingCodingExcersise
+{
+    /// <summary>
+    /// This program adds two user specified numbers utilizing a web client through a user specified port number.
+    /// </summary>
+    public class Program
+    {
+        /// <summary>
+        /// This represents the main logic of the program.
+        /// </summary>
+        /// <param name="args"></param>
+        public static void Main(string[] args)
+        {
+            int portNumber = DisplayMessageValidateInput();
+            ListenAsync(portNumber);
+            Console.WriteLine($"Server running on port {portNumber}. Press Enter to stop.");
+            Console.ReadLine();
+        }
+
+        /// <summary>
+        /// This method prompts the user to enter a port number, validates it, then returns it as an int..
+        /// </summary>
+        public static int DisplayMessageValidateInput()
+        {
+            int ConsoleValidatedInput = 0;
+            while (true)
+            {
+                Console.WriteLine("Enter a Port number to listen on:");
+                try
+                {
+                    int userInputPortNumber = Convert.ToInt32(Console.ReadLine());
+                    Console.WriteLine("Port number Accepted");
+                    ConsoleValidatedInput = userInputPortNumber;
+                    return ConsoleValidatedInput;
+                }
+                catch (FormatException)
```

Comment:
It would be preferable to  use int32.TryParse to determine whether the supplied  value is a valid integer rather than requiring an exception to be thrown for this since it is common for users to supply bad data.